extensions/zacoons/webgpu.js

```diff
@@ -0,0 +1,638 @@
+// Name: WebGPU
+// ID: webgpu
+// Description: WebGPU bindings for TurboWarp.
+// By: zacoons
+// License: CC0-1.0
+
+// todo
+// - add sample project after release
+// - add support for writing to MAP_WRITE buffers
+
+(function (Scratch) {
+  "use strict";
+
+  if (!Scratch.extensions.unsandboxed)
+    throw new Error("WebGPU extension must be run unsandboxed.");
+
+  function label(text) {
+    return {
+      blockType: Scratch.BlockType.LABEL,
+      text: text,
+    };
+  }
+
+  class GPUExtension {
+    constructor() {
+      Scratch.vm.on("PROJECT_RUN_STOP", () => {
+        this._resetAttrs();
+      });
+      this._resetAttrs();
+
+      this.types = {
+        Int8: Int8Array,
+        Uint8: Uint8Array,
+        Int16: Int16Array,
+        Uint16: Uint16Array,
+        Int32: Int32Array,
+        Uint32: Uint32Array,
+        Float32: Float32Array,
+        Float64: Float64Array,
+        Int64: BigInt64Array,
+        Uint64: BigUint64Array,
+      };
+    }
+
+    getInfo() {
+      return {
+        id: "webgpu",
+        name: "WebGPU",

```

Comment:
Since WebGPU is used for both rendering and compute, but this extension only provides compute functionality, maybe it's better to call this extension "WebGPU Compute"? Not sure, just an idea.

Reply:
I was gonna call it WebGPU bc I thought rendering functionality might be added eventually. But with your extension it seems unlikely that will happen, so I'll rename it. Nice job on Simple3D.

Reply:
> Nice job on Simple3D.

Thanks!

> But with your extension it seems unlikely that will happen

They aren't really competing though, since both are targeting different people with different goals. The only  competing extension targeting the same people is [my WebGL 2 bindings](https://github.com/TurboWarp/extensions/discussions/378) extension, but I'm not sure if I'll ever finish that.